app/models/skin.rb

```diff
@@ -498,6 +516,7 @@ def self.create_default
     end
     skin.official = true
     skin.save!
+    Rails.cache.increment('skins_generation/'+(skin.type.nil? ? ("site_skin") : skin.id.to_s) )
     skin
   end
   
```

Comment:
One final comment: I think this really could use an automated test to make sure the cache is expiring when you want it to. There's a test in bookmark_steps.rb that checks that the bookmark cache key changes appropriately -- taking that as a model, I would test in skin.feature that: 

- the style block cache doesn't expire when nothing has been changed
- the style block cache expires when the skin is edited
- the style block cache expires when the skin's parent is edited
- the style block cache expires when the skin's grandparent is edited
- the work skin style blocks don't expire when a site skin is edited